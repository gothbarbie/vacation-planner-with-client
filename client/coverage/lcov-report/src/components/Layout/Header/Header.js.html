<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/Layout/Header/Header.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../index.html">All files</a> / <a href="index.html">src/components/Layout/Header</a> Header.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">60% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>6/10</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>4/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">75% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>3/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">60% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>6/10</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">// @flow
&nbsp;
import React, { Component } from 'react'
import { connect } from 'react-redux'
import { Link, withRouter } from 'react-router-dom'
import type { MapStateToProps } from 'react-redux'
&nbsp;
import './Header.css'
import ButtonLink from '../../ButtonLink'
&nbsp;
type Props = {
  auth: {} | null,
  history: {
    location: string,
  },
}
&nbsp;
export class Header extends Component&lt;Props&gt; {
  renderLogo () {
    <span class="missing-if-branch" title="if path not taken" >I</span>if (this.props.auth) {
<span class="cstat-no" title="statement not covered" >      return (</span>
        &lt;Link to="/schedule"&gt;
          &lt;img
            alt="Vacation Planner"
            className="header__logo"
            src="/images/logo-v3.png"
          /&gt;
        &lt;/Link&gt;
      )
    }
    return (
      &lt;Link to="/"&gt;
        &lt;img
          alt="Vacation Planner"
          className="header__logo"
          src="/images/logo-v3.png"
        /&gt;
      &lt;/Link&gt;
    )
  }
&nbsp;
  renderLogin () {
    <span class="missing-if-branch" title="if path not taken" >I</span>if (this.props.auth) {
<span class="cstat-no" title="statement not covered" >      return (</span>
        &lt;ButtonLink default url="/api/logout"&gt;
          Logout
        &lt;/ButtonLink&gt;
      )
    }
    <span class="missing-if-branch" title="if path not taken" >I</span>if (this.props.history &amp;&amp; <span class="branch-1 cbranch-no" title="branch not covered" >!this.props.history.location === '/')</span> {
<span class="cstat-no" title="statement not covered" >      return (</span>
        &lt;ButtonLink primary url="/"&gt;
          Login
        &lt;/ButtonLink&gt;
      )
    }
  }
&nbsp;
  render () {
    return (
      &lt;header className="header"&gt;
        &lt;div className="header__inner"&gt;
          {this.renderLogo()}
          &lt;nav className="nav"&gt;
            &lt;Link to="/schedule"&gt;Schedule&lt;/Link&gt;
            &lt;span className="nav__separator" /&gt;
            &lt;Link to="/people"&gt;People&lt;/Link&gt;
            &lt;span className="nav__separator" /&gt;
            &lt;Link to="/destinations"&gt;Destinations&lt;/Link&gt;
          &lt;/nav&gt;
          {this.renderLogin()}
        &lt;/div&gt;
      &lt;/header&gt;
    )
  }
}
&nbsp;
const mapStateToProps = <span class="fstat-no" title="function not covered" >({</span> auth }) =&gt; {
<span class="cstat-no" title="statement not covered" >  return {</span>
    auth,
  }
}
&nbsp;
export default connect((mapStateToProps: MapStateToProps&lt;*, *, *&gt;))(
  withRouter(Header)
)
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Mar 11 2018 12:37:53 GMT+0100 (CET)
</div>
</div>
<script src="../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../sorter.js"></script>
</body>
</html>
