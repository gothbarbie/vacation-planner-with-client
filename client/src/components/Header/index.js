import React, { Component } from 'react'
import { connect } from 'react-redux'

type Props = {
  auth: {} | null
}

class Header extends Component<Props> {
  renderLogin () {
    if (this.props.auth) {
      return <a href="/api/logout">Logout</a>
    }
    return <a href="/auth/google">Login with Google</a>
  }

  render () {
    return (
      <header>
        <h1>Title</h1>
        <nav>
          {this.renderLogin()}
        </nav>
      </header>
    )
  }
}  

const mapStateToProps = ({ auth }) => {
  return {
    auth,
  }
}

export default connect((mapStateToProps: MapStateToProps<*, *, *>))(Header)
