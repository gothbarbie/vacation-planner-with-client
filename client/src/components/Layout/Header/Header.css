.header {
  background: #fff;
  box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.25);
  display: flex;
  justify-content: center;
}

.header__inner {
  width: 100%;
  max-width: 1140px;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  padding: .5rem;
}

.header__logo {

}

nav {
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
}
