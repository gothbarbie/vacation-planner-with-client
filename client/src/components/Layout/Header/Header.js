import React, { Component } from 'react'
import { connect } from 'react-redux'
import { Link, withRouter } from 'react-router-dom'

import './Header.css'
import ButtonLink from '../../ButtonLink'

type Props = {
  auth: {} | null,
  history: {
    location: string,
  }
}

export class Header extends Component<Props> {
  renderLogo () {
    if (this.props.auth) {
      return (
        <Link to="/schedule">
          <img
            alt="Vacation Planner"
            className="header__logo"
            src="/images/logo-v3.png"
          />
        </Link>
      )
    }
    return (
      <Link to="/">
        <img
          alt="Vacation Planner"
          className="header__logo"
          src="/images/logo-v3.png"
        />
      </Link>
    )
  }

  renderLogin () {
    if (this.props.auth) {
      return (
        <ButtonLink default url="/api/logout">
          Logout
        </ButtonLink>
      )
    }
    if (this.props.history && !this.props.history.location === '/') {
      return (
        <ButtonLink primary url="/">
          Login
        </ButtonLink>
      )
    }
  }

  render () {
    return (
      <header className="header">
        <div className="header__inner">
          {this.renderLogo()}
          <nav className="nav">
            <Link to="/schedule">Schedule</Link>
            <span className="nav__separator" />
            <Link to="/people">People</Link>
            <span className="nav__separator" />
            <Link to="/destinations">Destinations</Link>
          </nav>
          {this.renderLogin()}
        </div>
      </header>
    )
  }
}

const mapStateToProps = ({ auth }) => {
  return {
    auth,
  }
}

export default connect((mapStateToProps: MapStateToProps<*, *, *>))(
  withRouter(Header)
)
