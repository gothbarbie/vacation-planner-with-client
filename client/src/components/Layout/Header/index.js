import React, { Component } from 'react'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'

import './Header.css'
import { ButtonLink } from '../../common/Button'

type Props = {
  auth: {} | null,
}

class Header extends Component<Props> {
  renderLogin () {
    if (this.props.auth) {
      return (
        <ButtonLink default url="/api/logout">
          Logout
        </ButtonLink>
      )
    }
    return (
      <ButtonLink primary url="/auth/google">
        Login with Google
      </ButtonLink>
    )
  }

  render () {
    return (
      <header className="header">
        <div className="header__inner">
          <img className="header__logo" src="/images/logo-v3.png" />
          <nav className="nav">
            <Link to="/schedule">Schedule</Link>
            <span className="nav__separator" />
            <Link to="/people">People</Link>
            <span className="nav__separator" />
            <Link to="/destinations">Destinations</Link>
          </nav>
          {this.renderLogin()}
        </div>
      </header>
    )
  }
}

const mapStateToProps = ({ auth }) => {
  return {
    auth,
  }
}

export default connect((mapStateToProps: MapStateToProps<*, *, *>))(Header)
