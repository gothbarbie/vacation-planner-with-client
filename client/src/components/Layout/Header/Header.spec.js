import React from 'react'
import { shallow } from 'enzyme'

import { Header } from './Header'

describe('components/Layout/Header', () => {
  const props = {
    auth: null,
  }
  const header = shallow(<Header {...props} />)

  it('renders', () => {
    expect(header).toMatchSnapshot()
  })
})
