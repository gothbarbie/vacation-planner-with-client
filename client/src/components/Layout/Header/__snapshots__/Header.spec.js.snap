// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Layout/Header renders 1`] = `
<header
  className="header"
>
  <div
    className="header__inner"
  >
    <Link
      replace={false}
      to="/"
    >
      <img
        alt="Vacation Planner"
        className="header__logo"
        src="/images/logo-v3.png"
      />
    </Link>
    <nav
      className="nav"
    >
      <Link
        replace={false}
        to="/schedule"
      >
        Schedule
      </Link>
      <span
        className="nav__separator"
      />
      <Link
        replace={false}
        to="/people"
      >
        People
      </Link>
      <span
        className="nav__separator"
      />
      <Link
        replace={false}
        to="/destinations"
      >
        Destinations
      </Link>
    </nav>
  </div>
</header>
`;
